include/boost/geometry/strategies/spherical/distance_cross_track_point_box.hpp

```diff
@@ -32,131 +44,143 @@ namespace boost { namespace geometry
-            return_type const crs_AB = geometry::detail::course<return_type>(p1, p2);
-            return_type const d_crs1 = crs_AD - crs_AB;
-            return_type const sin_d_crs1 = sin(d_crs1);
-
-            // this constant sin() is here to be consistent with the side strategy
-            return_type const sigXTD = asin(sin(0.001) * sin_d_crs1);
-
-            // If the point is on the right side of the edge
-            if ( sigXTD > 0 )
+            if (plat > lat_max)
+            {
+                return services::result_from_distance
+                    <
+                        Strategy, Point, box_point_type
+                    >::apply(m_ps_strategy, radius() * (plat - lat_max));
+            }
+            else if (plat < lat_min)
             {
-                return_type const crs_BA = crs_AB - geometry::math::pi<return_type>();
-                return_type const crs_BD = geometry::detail::course<return_type>(p2, point);
-                return_type const d_crs2 = crs_BD - crs_BA;
-
-                return_type const projection1 = cos( d_crs1 );
-                return_type const projection2 = cos( d_crs2 );
-
-                if(projection1 > 0.0 && projection2 > 0.0)
-                {
-                    return_type const d1 = m_pp_strategy.apply(p1, point);
-                    return_type const
-                        XTD = radius()
-                            * geometry::math::abs(
-                                asin( sin( d1 / radius() ) * sin_d_crs1 )
-                              );
-
-                    return return_type(XTD);
-                }
-                else
-                {
-                    // OPTIMIZATION
-                    // Return d1 if projection1 <= 0 and d2 if projection2 <= 0
-                    // if both == 0 then return d1 or d2
-                    // both shouldn't be < 0
-
-                    return_type const d1 = m_pp_strategy.apply(p1, point);
-                    return_type const d2 = m_pp_strategy.apply(p2, point);
-
-                    return return_type((std::min)( d1 , d2 ));
-                }
+                return services::result_from_distance
+                    <
+                        Strategy, Point, box_point_type
+                    >::apply(m_ps_strategy, radius() * (lat_min - plat));
+            }
+            else
+            {
+                BOOST_GEOMETRY_ASSERT(plat >= lat_min && plat <= lat_max);
+                return return_type(0);
             }
         }
 
-        // Return 0 if the point isn't on the right side of any segment
-        return return_type(0);
+        // Otherwise determine the distance of the point to the box as
+        // the minimum distance of the point to the western- and
+        // eastern-most edges of the box
+        return_type d1 = m_ps_strategy.apply(point, bottom_left, top_left);
+        return_type d2 = m_ps_strategy.apply(point, bottom_right, top_right);
+
+        return (std::min)(d1, d2);
```

Comment:
Yes, you're right. The strategy must be called in all cases since the shortest path from point to the meridian will be a fragment of a great circle, and won't go along a parallel.